ciao-controller/client.go

```diff
@@ -73,6 +73,21 @@ func (client *ssntpClient) EventNotify(event ssntp.Event, frame *ssntp.Frame) {
 			glog.Warning("Error unmarshalling InstanceDeleted")
 			return
 		}
+
+		attachments, _ := client.ctl.ds.GetStorageAttachments(event.InstanceDeleted.InstanceUUID)
+		for _, attachment := range attachments {
+			if attachment.Ephemeral {
+				err = client.ctl.ds.DeleteBlockDevice(attachment.BlockID)
+				if err != nil {
+					glog.Warningf("Error deleting block device from datastore: %v", err)
+				}
+				err = client.ctl.DeleteBlockDevice(attachment.BlockID)
+				if err != nil {
+					glog.Warningf("Error deleting block device: %v", err)
+				}
+			}
+		}
+
 		client.ctl.ds.DeleteInstance(event.InstanceDeleted.InstanceUUID)
```

Comment:
Sorted by removing the attachment added which then makes ds.deleteInstance happy.